core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/machineLearning/sklearnAdvanced/base/SklearnAdvancedBaseDesc.scala

```diff
@@ -0,0 +1,154 @@
+  @JsonPropertyDescription("Ground truth attribute column")
+  @AutofillAttributeName
+  var groundTruthAttribute: String = ""
+
+  @JsonProperty(value = "Selected Features", required = true)
+  @JsonSchemaTitle("Selected Features")
+  @JsonPropertyDescription("Features used to train the model")
+  @AutofillAttributeNameList
+  var selectedFeatures: List[String] = _
+
+  private def getLoopTimes(paraList: List[HyperParameters[T]]): String = {
+    for (ele <- paraList) {
+      if (ele.parametersSource) {
+        return s"""table[\"${ele.attribute}\"].values.shape[0]"""
+      }
+    }
+    "1"
+  }
+
+  def getParameter(paraList: List[HyperParameters[T]]): List[String] = {
+    var str1 = ""; var str2 = ""; var str3 = ""
+    for (ele <- paraList) {
+      if (ele.parametersSource) {
+        str1 = str1 + String.format("%s = {},", ele.parameter.getName)
+        str2 =
+          str2 + String.format("%s(table['%s'].values[i]),", ele.parameter.getType, ele.attribute)
+        str3 = str3 + String.format(
+          "%s = %s(table['%s'].values[i]),",
+          ele.parameter.getName,
+          ele.parameter.getType,
+          ele.attribute
+        )
+      } else {
+        str1 = str1 + String.format("%s = {},", ele.parameter.getName)
+        str2 = str2 + String.format("%s ('%s'),", ele.parameter.getType, ele.value)
+        str3 = str3 + String.format(
+          "%s = %s ('%s'),",
+          ele.parameter.getName,
+          ele.parameter.getType,
+          ele.value
+        )
+      }
+    }
+    List(String.format("\"%s\".format(%s)", str1, str2), str3)
+  }
+
+  override def generatePythonCode(): String = {
+    val listFeatures = selectedFeatures.map(feature => s""""$feature"""").mkString(",")
+    val trainingName = getImportStatements.split(" ").last
+    val stringList = getParameter(paraList)
+    val trainingParam = stringList(1)
+    val paramString = stringList(0)
+    val finalCode =
+      s"""
+         |from pytexera import *
+         |
+         |import pandas as pd
+         |${getImportStatements}
+         |
+         |class ProcessTableOperator(UDFTableOperator):
+         |
+         |  @overrides
+         |  def process_table(self, table: Table, port: int) -> Iterator[Optional[TableLike]]:
+         |    global dataset
+         |    model_list = []
+         |    para_list = []
+         |    features = [$listFeatures]
+         |
+         |    if port == 0:
+         |      dataset = table
+         |
+         |    if port == 1 :
+         |      y_train = dataset["$groundTruthAttribute"]
+         |      X_train = dataset[features]
+         |      loop_times = ${getLoopTimes(paraList)}
+         |
+         |      for i in range(loop_times):
+         |        model = ${trainingName}(${trainingParam})
+         |        model.fit(X_train, y_train)
+         |        model_list.append(model)
+         |        para_str = ${paramString}
+         |        para_list.append(para_str)
```

Comment:
I still suggest making each operator simple. training operator trains a model with a given input. for-loop should be done by control blocks, not internally to the training operator. If this is a temporary solution I am fine, but please leave comments here for the future plan and todos.

Reply:
That is a temporary solution, and we will modify it after control block merging into master.

Reply:
ok